### ISSUE TITLE
game keeps crashing after recent patch, want to try a repair.. but how?

### ISSUE THREAD

**Issue by zman112** [Healix/Gw2Launcher#371]
Hello,

I've been crashing more than normal (normal being less than 1 crash/month before this September patch). Now I'm crashing every few days.

many crashes seem to have the same error:
Assertion: bgfxUniformsCount == shader.constantsCount + shader.samplersCount
File: D:\Perforce\Live\NAEU\v2\Code\Arena\Engine\Gr\Bgfx\BgfxShader.cpp(191)

once it was this error:
Assertion: Existing cube data for file 0x1aec77 already loaded with flags 0, but trying to load with flags 16 now.
File: D:\Perforce\Live\NAEU\v2\Code\Arena\Engine\Gr\GrTexCube.cpp(970)

I'm not saying it has anything to do with gw2launcher, I have no idea how to begin troubleshooting.

I'm thinking about trying to do a game repair but I'm not sure how to go about that since I have many accounts and I use gw2launcher to run the game. Does doing the repair via Gw2-64.exe as instructed by Anet repair all the account's files?

Any advice?

Thank you!

**Comment by Healix** [comment]
> Does doing the repair via Gw2-64.exe as instructed by Anet repair all the account's files?

If the accounts are using the default Local.dat file, yes, otherwise that's the only file it wouldn't repair.

To repair, add -repair to the arguments under Settings > Guild Wars 2 > Launch options. Right click the account > Selected > Launch (normal). It should begin repairing. Remove the option when you're done.

**Comment by zman112** [comment]
Thank you Healix, I did the repair and am still crashing, same shader error... 

I searched all over and seems like there's one only 1 recent report of this exact "BgfxShader.cpp(191)" error https://en-forum.guildwars2.com/topic/137985-game-keep-crashing-at-cursed-shore/

I want to try it, but I'm unsure how to properly clear cache when gw2launcher is involved... I see there's a folder seemingly for every account I have set up, but they're all numbered folders and I can't tell which folder number corresponds to which account, and it's confusing me because:
-some numbered folders are empty
-the numbers skip (there's no 12-25)
-some folders look like shortcuts
-inside some numbered folders, there's more appdata/roaming/gw2/coherent dumps/[more numbered folders]

so for example... if I look at account#7? or whatever... there's more appdata within it that leads to more numbered folders, and most of them have gw2cache:
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\7\AppData\Roaming\Guild Wars 2\Coherent Dumps\2\gw2cache-{AC00A8C1-C343-2ACC-C2A8-00AC43C3CC2A}
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\7\AppData\Roaming\Guild Wars 2\Coherent Dumps\3\gw2cache-{AC00A8C1-C352-2ACC-C2A8-00AC52C3CC2A}
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\7\AppData\Roaming\Guild Wars 2\Coherent Dumps\36\gw2cache-{AC00A8C1-D5E9-2ACC-C2A8-00ACE9D5CC2A}
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\7\AppData\Roaming\Guild Wars 2\Coherent Dumps\37\gw2cache-{AC00A8C1-D606-2ACC-C2A8-00AC06D6CC2A}
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\7\AppData\Roaming\Guild Wars 2\Coherent Dumps\39\gw2cache-{AC00A8C1-D647-2ACC-C2A8-00AC47D6CC2A}

Then if I go to folder 9, instead of 7... it's the same thing:
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\9\AppData\Roaming\Guild Wars 2\Coherent Dumps\2\gw2cache-{AC00A8C1-C343-2ACC-C2A8-00AC43C3CC2A}
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\9\AppData\Roaming\Guild Wars 2\Coherent Dumps\3\gw2cache-{AC00A8C1-C352-2ACC-C2A8-00AC52C3CC2A}

If I go to the same number folder both both, it's empty...
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\7\AppData\Roaming\Guild Wars 2\Coherent Dumps\7\[empty]
...or NOT empty
C:\Users\xx\AppData\Roaming\Gw2Launcher\data\27\AppData\Roaming\Guild Wars 2\Coherent Dumps\27\gw2cache-{AC00A8C1-C175-2ACC-C2A8-00AC75C1CC2A}
maybe there's no correlation.

Some folders only has "gw2cache-user" and "gw2cache" but no "gw2cache-{AC00A8C1..."

Makes me feel like I'm computer illiterate, I don't know what any of this means. I wonder if I set everything up incorrectly from the beginning and worth starting over

**Comment by Healix** [comment]
Right click any account > Tools > Advanced > Delete cache folders. Click delete cache. This will delete all gw2cache folders, including those outside of Gw2Launcher.

Updating will also delete gw2cache. All > Update to delete it for all accounts or Selected > Update to delete it for specific accounts.

"Coherent Dumps" is a link that goes back to %appdata%\Gw2Launcher\data\temp, so those # folders are actually under the temp folder. There will be a # folder per account, which is the ID of the account, and the contents of each folder will look similar, as those are the temporary files that GW2 creates on every launch. Files in the temp folder are automatically deleted after a week if GW2 hasn't been using them.

"gw2cache" is a remembered cache location, allowing the cache to be reused. You'll only see this when the option for it is enabled (Settings > Tools > Local.dat > Static cache location).

"gw2cache-user" is used when localized execution set to binaries is enabled. It's the gw2cache\user folder.

"gw2cache-{...}" is what GW2 normally uses, where {...} is randomly generated.

**Comment by zman112** [comment]
If updating deletes gw2cache? Wouldn't that mean my cache was deleted, as recently as today? There was a small update this afternoon.

**Comment by Healix** [comment]
Yes, if Gw2Launcher did the update, the gw2cache folders should have been deleted and then recreated after the update.
